Std/Data/BitVec/Lemmas.lean

```diff
@@ -499,3 +511,42 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   let ⟨y, lt⟩ := y
   simp
   exact Nat.lt_of_le_of_ne
+
+-- YESUPSTREAMABLE
+@[simp]
+theorem BitVec.ofInt_ofNat : BitVec.ofInt w (no_index (OfNat.ofNat n : Int)) = BitVec.ofNat w n := by
+  simp [BitVec.ofInt]
+
+-- YESUPSTREAMABLE.
+-- for completeness, also do the signed versions with `toInt` and the other unsigned versions.
+theorem BitVec.ult_ofNat (w : Nat) (x y : Nat)  :
+    (BitVec.ult (BitVec.ofNat w x) (BitVec.ofNat w y)) = decide (x % 2^w < y % 2^w) := by
+  simp [BitVec.ult]
+  constructor
+  · simp [BitVec.ofNat, BitVec.toFin, Fin.ofNat']
+  · intros h
+    simp only [BitVec.ofNat, Fin.ofNat', Fin.mk_lt_mk, h]
```

Comment:
Agreed, this should be uncontroversial

Reply:
https://github.com/opencompl/std4/pull/24